### PR TITLE
feat(gatsby-plugin-preload-fonts): remove domain for self-hosted fonts

### PR DESCRIPTION
**Pull request by universse**
## Description
Remove domain from self-hosted fonts' urls for ```gatsby-plugin-preload-fonts```.

### CODE REVIEW

**Comment by wardpeet** [comment]
Hey universse!

Thanks for this PR! We really appreciate you taking the time to open it.

In the interest of moving this along most effectively, we strongly encourage our contributors to provide an example project which we can then use to validate the changes in this PR. This will allow us to quickly and easily ensure that this does indeed address the purported feature or fix in this PR!

An example project:

can be run locally or credentials have been provided
Feel free to message on Twitter, Discord, etc.
can be used to validate this functionality
In other words, the necessary functionality has been enabled/disabled in the project for us to easily validate these > changes>
If you're able to provide an example that we can use to test these changes we can more quickly get this merged and released. If you're unable--we will prioritize as best as we can and test when we're able to do so.

Thanks for this PR and thanks for using Gatsby!

💜

**Comment by universse** [comment]
I'm not really sure how to set up a example project showing the plugin's new changes.

**Comment by universse** [comment]
I managed to set up the example project. Here's the output.

When running preload-fonts script:
image

font-preload-cache.json (formatted)
image

html output
image

Tests are failing though and I'm not sure how to fix that.

**Comment by LekoArts** [comment]
> I managed to set up the example project.

Can you share this project with us please? Thanks!

**Comment by universse** [comment]
> 
> 
> > I managed to set up the example project.
> 
> Can you share this project with us please? Thanks!

Yep, probably by weekend. The above screenshots were taken in a local project where I directly modified gatsby from node_modules. Is that sufficient?

**Comment by universse** [comment]
Here it is https://github.com/universse/gatsby-preload-fonts

File: packages/gatsby-plugin-preload-fonts/src/gatsby-ssr.js [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
   const assets = Object.keys(cache.assets[pathname])
+
   setHeadComponents(
-    assets.map(href => (
-      <link key={href} rel="preload" href={href} as="font" {...props} />
-    ))
+    assets.map(href => {
+      let assetProps
+
+      try {
```
[the remark this answers is not in the document]
Comment (universse) [reply packages/gatsby-plugin-preload-fonts/src/gatsby-ssr.js]: I guess it's to override props crossOrigin from config option, which only applies to external fonts. If that's okay I will push a commit with comment soon.

**Comment by universse** [comment]
Thanks for reviewing. It's my pleasure :sparkles::sparkles:

**Pull request merged**